server/container/spring/src/main/java/org/apache/james/container/spring/bean/factorypostprocessor/MailboxConfigurationBeanFactoryPostProcessor.java

```diff
@@ -96,10 +112,21 @@ public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
             registry.registerAlias(mailboxIdDeserializer, "mailbox-id-deserializer");
             registry.registerAlias(mailboxIdFactory, "mailboxIdFactory");
 
+            removeMailboxManagersExceptRightSelectedOne(registry, mailbox);
+
         } catch (ConfigurationException e) {
             throw new FatalBeanException("Unable to config the mailboxmanager", e);
         }
 
     }
 
+    private void removeMailboxManagersExceptRightSelectedOne(BeanDefinitionRegistry registry, String selectedMailboxManager) {
```

Comment:
I would have done this the other way:
keep the information of what implementation has already been loaded in a set, and don't load it if it's already present in this set.